modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -754,49 +762,72 @@ void ONNXImporter::parseOperatorSet()
 }
 
-void ONNXImporter::handleQuantizedNode(LayerParams& layerParams,
-                                       const opencv_onnx::NodeProto& node_proto)
+static bool hasInt8Input(const String& inputLayerType)
+{
+    static std::vector<String> int8ListLayerType = {
+            "Quantize",
+            "Requantize",
+            "ConvolutionInt8",
+            "InnerProductInt8",
+            "PoolingInt8",
+            "ReduceInt8",
+            "EltwiseInt8",
+            "BatchNormInt8",
+            "ScaleInt8",
+            "ShiftInt8",
+            "ReLUInt8",
+            "ReLU6Int8",
+            "SigmoidInt8",
+            "TanHInt8",
+            "SwishInt8",
+            "MishInt8",
+            "ELUInt8",
+            "BNLLInt8",
+            "AbsValInt8",
+            "SoftmaxInt8",
+            "SoftMaxInt8",
+            "ConcatInt8",
+            "FlattenInt8",
+            "PaddingInt8",
+            "BlankInt8",
+            "DropoutInt8",
+            "IdentityInt8",
+            "SilenceInt8",
+            "ConstInt8",
+            "ReshapeInt8",
+            "ResizeInt8",
+            "SplitInt8",
+            "SliceInt8",
+            "CropInt8",
+            "PermuteInt8",
+            "ReorgInt8",
+            "ShuffleChannelInt8",
+            };
+    auto layerIt = std::find(int8ListLayerType.begin(), int8ListLayerType.end(), inputLayerType);
+    return layerIt != int8ListLayerType.end();
+}
+
+void ONNXImporter::setParamsDtype(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto)
 {
-    // Quantized nodes have output names ending with 'quantized'
-    std::string outName = node_proto.output(0);
-    int len = outName.length();
-    if (len <= 9)
-        return;
+    // If the output format of the previous layer is int8, than we think the current
+    // layer should also output int8.
 
-    if (outName.substr(len - 9) == "quantized")
+    // Check if the layer has a variable of Int8 as input.
+    for (int i = 0; i < node_proto.input_size(); ++i)
     {
-        outName = outName.substr(0, len - 9);
-        Mat scale, zeropoint;
-
-        if (constBlobs.find(outName + "scale") != constBlobs.end() &&
-            constBlobs.find(outName + "zero_point") != constBlobs.end())
-        {
-            scale = getBlob(outName + "scale");
-            zeropoint = getBlob(outName + "zero_point");
-        }
-        else
+        if (layer_id.find(node_proto.input(i)) != layer_id.end())
         {
-            std::string inpName = node_proto.input(0);
-            inpName = inpName.substr(0, inpName.length() - 9);
-            scale = getBlob(inpName + "scale");
-            zeropoint = getBlob(inpName + "zero_point");
+            LayerInfo layerInfo = layer_id.find(node_proto.input(i))->second;
+            Ptr<Layer> preLayer = dstNet.getLayer(layerInfo.layerId);
 
-            for (int i = 0; i < node_proto.output_size(); i++)
+            if (layerInfo.depth == CV_8S && hasInt8Input(preLayer->type))
```

Comment:
So if we insert unsupported at the moment layer(Gather, for example) after integer node, it will set the depth to CV_8S and corrupt the memory at runtime? (reading int8 as float in this case).

Reply:
What if we have a set of ONNX layers instead? e.g. in this set we would put `QuantizeLinear`, `DequantizeLinear`, `QLinearConv`, `Transpose`, `Reshape`, `MaxPool`, etc.
And check that if we have integer inputs, current layer should be in this set, otherwise we throw an error, what do you think?

Reply:
> So if we insert unsupported at the moment layer(Gather, for example) after integer node, it will set the depth to CV_8S and corrupt the memory at runtime? (reading int8 as float in this case).

Yes, if we do this, it will corrupt. But this should not happen, Gather-like layer should output the same format as the input. So, if a Gather layer gets int8 input, it should be `GatherInt8`.

Reply:
> What if we have a set of ONNX layers instead? e.g. in this set we would put QuantizeLinear, DequantizeLinear, QLinearConv, Transpose, Reshape, MaxPool, etc.
And check that if we have integer inputs, the current layer should be in this set, otherwise we throw an error, what do you think?

I also considered a list of ONNX operators. Functionally, there is no difference between the two lists in essence, one is the original version of ONNX and the other is the mapped version of OpenCV.
Considering the future maintainability, indeed using the original version of ONNX operator names is better.